crates/supervisor/core/src/rpc/server.rs

```diff
@@ -0,0 +1,95 @@
+//! Server-side implementation of the Supervisor RPC API.
+
+use crate::supervisor::SupervisorService;
+use alloy_eips::eip1898::BlockNumHash;
+use alloy_primitives::{B256, ChainId};
+use async_trait::async_trait;
+use jsonrpsee::{
+    core::RpcResult,
+    types::{ErrorObject, error::ErrorCode},
+};
+use kona_interop::{DerivedIdPair, ExecutingDescriptor, SafetyLevel, SuperRootResponse};
+use kona_supervisor_rpc::SupervisorApiServer;
+use std::sync::Arc;
+use tracing::{trace, warn};
+
+/// The server-side implementation struct for the `SupervisorApi`.
+/// It holds a reference to the core Supervisor logic.
+#[derive(Debug, Clone)]
+pub struct SupervisorRpc {
+    /// Reference to the core Supervisor logic.
+    /// Using Arc allows sharing the Supervisor instance if needed,
+    supervisor: Arc<dyn SupervisorService + Send + Sync + 'static>,
```

Comment:
no need to dynamic dispatch here, let's use and unbounded a generic with a default instead and trait bound the generic on the impl bodies which need it